apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/components/RubricItemsFilter.tsx

```diff
@@ -0,0 +1,186 @@
+import { useMemo } from 'preact/compat';
+import { Dropdown } from 'react-bootstrap';
+
+import type { RubricData } from '../../../../lib/manualGrading.types.js';
+import type { InstanceQuestionRow } from '../assessmentQuestion.types.js';
+
+interface RubricItemsFilterProps {
+  rubricData: RubricData | null;
+  instanceQuestions: InstanceQuestionRow[];
+  rubricItemsFilter: string[];
+  setRubricItemsFilter: (value: string[] | ((prev: string[]) => string[])) => Promise<unknown>;
+}
+
+export function RubricItemsFilter({
+  rubricData,
+  instanceQuestions,
+  rubricItemsFilter,
+  setRubricItemsFilter,
+}: RubricItemsFilterProps) {
+  // Calculate usage statistics for each rubric item
+  const rubricItemUsage = useMemo(() => {
+    if (!rubricData) return new Map<string, number>();
+    const usage = new Map<string, number>();
+
+    // Initialize all items to 0
+    rubricData.rubric_items.forEach((item) => {
+      usage.set(item.id, 0);
+    });
```

Comment:
Why is initialization needed, doesn't `usage.get(itemId) || 0` below handle this?